{{cookiecutter.project_slug}}/containers/Dockerfile

```diff
@@ -0,0 +1,82 @@
+# ref: https://github.com/mamba-org/micromamba-docker/blob/main/Dockerfile
+{% if cookiecutter.add_containers == "Docker" %}
+FROM condaforge/mambaforge:latest
+{% else %}
+FROM docker.io/condaforge/mambaforge:latest
+{% endif %}
+LABEL maintainer="{{ cookiecutter.author_full_name }} <{{ cookiecutter.author_email }}>"
+LABEL org.opencontainers.image.title="{{ cookiecutter.project_name }}"
+LABEL org.opencontainers.image.authors="{{ cookiecutter.project_name }} Team"
+LABEL org.opencontainers.image.source="{{ cookiecutter.git_https_origin }}"
+LABEL org.opencontainers.image.version="latest"
+LABEL org.opencontainers.image.description="{{ cookiecutter.project_short_description }}"
+
+# it is the default, but using it here to have it explicitly
+USER root
+
+{% if cookiecutter.add_containers == "Docker" -%}
+SHELL ["/bin/bash", "-c"]
+{%+ endif -%}
+# Use bash in Dockerfile RUN commands and make sure bashrc is sourced when
+# executing commands with /bin/bash -c
+# Needed to have the micromamba activate command configured etc.
+
+ENV ENV_NAME={{ cookiecutter.package_slug }}
+{% if cookiecutter.add_containers == "Docker" -%}
```

Comment:
sorry could you explain .. why it is necessary to check if it uses docker or podman 
in all the file below?

and why it uses micromamba for podman?

Reply:
I used https://www.perplexity.ai/ to view what changes are needed to the Dockerfile to make it work with Podman and it showed me the following:

> The given Dockerfile can be adapted to work with Podman. Here are the changes that need to be made:
> 
> - Change the FROM command to use the fully qualified image name, including the registry. For example, docker.io/condaforge/mambaforge:latest.
> - Remove the LABEL commands that specify the maintainer, source, and authors, since they are not required by Podman.
> - Remove the USER command, since Podman does not require root privileges to run containers.
> - Change the RUN command that installs packages to use dnf instead of apt-get, since Podman uses dnf as its package manager.
> - Change the CMD command to use micromamba instead of python, since the micromamba command is used to activate the virtual environment.

I was also seeing that a Dockerfile can work in both cases, the difference that may be more noticeable is that some permissions are not needed in Podman

Reply:
I think that it is ok to change FROM .. and maybe LABEL .. but the other ones I think that it doesn't make too much sense. so for now I would keep them ... and maybe if someone complain about that we can get feedback and add something more specific for podam (but I hope it will not be necessary)